src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/QueueListenerImpl.java

```diff
@@ -61,6 +62,10 @@ public void onLeft(Queue.LeftItem li) {
             json.put(Util.KEY_DEQUEUE_REASON, Util.VALUE_CANCELLED);
         } else {
             json.put(Util.KEY_DEQUEUE_REASON, Util.VALUE_BUILDING);
+            Label assignedLabel = li.getAssignedLabel();
+            json.put(Util.KEY_DEQUEUE_ALLOCATED_LABEL,
+                assignedLabel != null ? assignedLabel.getDisplayName() : Util.KEY_DEQUEUE_NO_LABEL);
```

Comment:
True. From a data modelling perspective I'm not fond of magic strings like NO_LABEL but from a practical point of view it's... practical. While you can search for documents that don't have a particular field you can't get it into an aggregation. I'm fine with either.